### PR TITLE
Sb 24483 Observation consumption flow 

### PR DESCRIPTION
**Pull request by aman-tunerlabs**

### CODE REVIEW

File: controllers/v2/observationsController.js [hunk 1]
```diff
@@ -835,13 +839,28 @@ module.exports = class Observations extends v1Observation {
...
                     submissionDocument
                 );
 
                 assessment.submissionId = submissionDoc.result._id;
 
+                if( req.query.ecmMethod && req.query.ecmMethod !== "" ) {
+                    
+                    if(evidenceMethodArray[req.query.ecmMethod]){
+                        let selectedEvidence = evidenceMethodArray[req.query.ecmMethod]
+                        evidenceMethodArray  = [];
```
Comment (aman-tunerlabs) [inline controllers/v2/observationsController.js:854]: Rakesh why are we making evidenceMethodArray = []

File: controllers/v2/observationsController.js [hunk 2]
```diff
@@ -835,13 +839,28 @@ module.exports = class Observations extends v1Observation {
...
+                        evidenceMethodArray.push(selectedEvidence);
+                    } else {
+                        return resolve({
```
Comment (aman-tunerlabs) [inline controllers/v2/observationsController.js:857]: Rakesh why observation submission not found message is thrown. Isn't it ecm is invalid and should be a bad request ?

File: controllers/v2/observationsController.js (hunk 1 above)
Comment (aman-tunerlabs) [reply controllers/v2/observationsController.js:854]: Rakesh evidenceMethodArray is it array in existing flow ??

File: module/observationSubmissions/helper.js [hunk 3]
```diff
@@ -502,6 +506,26 @@ module.exports = class ObservationSubmissionsHelper {
                 resultedData.submissionDate = resultedData.completedDate ? resultedData.completedDate : "";
                 resultedData.ratingCompletedAt = resultedData.ratingCompletedAt ? resultedData.ratingCompletedAt : "";
 
+                let domains = []; if(resultedData.evidencesStatus){
```
Comment (aman-tunerlabs) [inline module/observationSubmissions/helper.js:509]: Rakesh In which scenario evidencesStatus won't be there for observation submissions ??

File: module/observationSubmissions/helper.js [hunk 4]
```diff
@@ -502,6 +506,26 @@ module.exports = class ObservationSubmissionsHelper {
...
+                        } 
+                        domains.push({ 
+                            name: evidence.name,
+                            code: evidence.externalId, 
+                            status: status 
+                        }) 
+                    }); 
+                }
+                resultedData['domains'] =domains; 
+                delete resultedData.evidencesStatus;
```
Comment (aman-tunerlabs) [inline module/observationSubmissions/helper.js:527]: Rakesh how will it work once deleted from the object ?

File: module/observationSubmissions/helper.js (hunk 3 above)
Comment (rakeshSgr) [reply module/observationSubmissions/helper.js]: I'm not sure, so I have added the condition before getting the data.

File: controllers/v2/observationsController.js [hunk 5]
```diff
@@ -835,13 +839,30 @@ module.exports = class Observations extends v1Observation {
...
                 submissionDocument.submissionNumber = submissionNumber;
+            
 
                 let submissionDoc = await observationsHelper.findSubmission(
                     submissionDocument
                 );
 
                 assessment.submissionId = submissionDoc.result._id;
 
+                if( req.query.ecmMethod && req.query.ecmMethod !== "" ) {
```
Comment (aks30) [inline controllers/v2/observationsController.js:850]: @rakeshSgr , @aman-tunerlabs this condition can be checked earlier before creating submission document for the first time, all evidence for the first call will be put in submission document.

Comment (aks30) [reply controllers/v2/observationsController.js:850]: @rakeshSgr - Add this code in v1 as well and in assessments too if its not too complicated.

File: models/observations.js [hunk 6]
```diff
@@ -50,6 +50,8 @@ module.exports = {
       index: true
     },
     project : Object,
-    referenceFrom : String
+    referenceFrom : String,
+    language: Array,
```
Comment (aks30) [inline models/observations.js:54]: @rakeshSgr , @aman-tunerlabs - Please get the same value from solution document and not put it here.

File: module/observations/helper.js [hunk 7]
```diff
@@ -1629,7 +1629,9 @@ module.exports = class ObservationsHelper {
                         "name" : 1, 
                         "description" : 1,
                         "solutionId" : 1,
-                        "programId" : 1
+                        "programId" : 1,
+                        "language" : 1,
```
Comment (aks30) [inline module/observations/helper.js:1633]: @aman-tunerlabs , @rakeshSgr as discussed not required here

File: controllers/v2/observationsController.js (hunk 5 above)
Comment (rakeshSgr) [reply controllers/v2/observationsController.js:860]: Its fixed and updated in assessment/details API and observations/v1 as well

**Pull request merged**